### PR TITLE
Read-Only backend for Pricing in genomics status

### PR DESCRIPTION
**Pull request by alneberg**
This together with https://github.com/SciLifeLab/standalone_scripts/pull/53 can reproduce the numbers for prices available in the excel sheet currently used (1378 Cost Calculator). It is read-only at this stage but the plan is to extend it with write functionality next.

The basic logic is that an order consists of different products which in turn consists of a number of components, so:
ORDERS -> PRODUCTS -> COMPONENTS

However, the order level is not yet implemented.

These are the URL:s implemented (GET):
```
            /api/v1/pricing_date_to_version
            /api/v1/pricing_components
            /api/v1/pricing_components/<ID>
            /api/v1/pricing_exchange_rates
            /api/v1/pricing_products
            /api/v1/pricing_products/<ID>
```

### CODE REVIEW

**Comment by alneberg** [comment]
Now live on stage. Ping @remiolsen

File: status/pricing.py [hunk 1]
```diff
@@ -0,0 +1,398 @@
...
+                        "individual_products/by_id_and_version",
+                        key=[int_key, int_version],
+                        limit=1,
+                        descending=True
+                    )
+        else:  # No specified version
+            if product_id is None:  # All products
+                view = self.application.pricing_products_db.view(
+                        "entire_document/by_version",
+                        limit=1,
```
Comment (remiolsen) [inline status/pricing.py]: If you want to get several rows here, wouldn't `limit=1` be counter to that?

File: status/pricing.py [hunk 2]
```diff
@@ -0,0 +1,398 @@
...
+                        descending=True
+                    )
+            else:
+                view = self.application.pricing_products_db.view(
+                        "individual_products/by_id_and_version",
+                        startkey=[int_key, {}],
+                        endkey=[int_key],
+                        limit=1,
```
Comment (remiolsen) [inline status/pricing.py]: See above comment.

File: status/pricing.py [hunk 3]
```diff
@@ -0,0 +1,398 @@
...
+
+    def _validate_version_param(self, version):
+        try:
+            int_version = int(version)
+        except ValueError:
+            raise tornado.web.HTTPError(
+                400, reason='Bad request, version is not an integer')
+        return int_version
+
+    def _validate_object_id(self, id, object_type):
```
[the remark this answers is not in the document]
Comment (alneberg) [reply status/pricing.py]: YES! Especially how we both became authors of that one commit! 👍

**Pull request merged**